### PR TITLE
Add apr calculation

### LINKED ISSUE #43: Add APR calculations from API

### PR DESCRIPTION
**Pull request by ghost**
### Description of changes proposed in this pull request:
- Implement apr calculation

### Checks:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: src/utils/MarketUtils.ts [hunk 1]
```diff
@@ -337,3 +351,341 @@ export const getDailyPriceHistory = async (group: string, synthMetadata: Record<
...
+    }
+
+    // @notice New calculation based on the doc
+    // umaRewardsPercentage = (`totalTokensOutstanding` * synthPrice) / whitelistedTVM
+    let umaRewardsPercentage = BigNumber.from(collateralCount.toString()).mul(synthTokenPrice);
+    umaRewardsPercentage = umaRewardsPercentage.div(tokenCount.toString());
+    // dynamicAmountPerWeek = 50,000 * umaRewardsPercentage
+    const dynamicAmountPerWeek = umaRewardsPercentage.mul(umaRewards);
+    // dynamicAmountPerWeekInDollars = dynamicAmountPerWeek * UMA price
+    const dynamicAmountPerWeekInDollars = dynamicAmountPerWeek.mul(umaPrice);
```
Comment (ghost) [inline src/utils/MarketUtils.ts]:
```
error Error: underflow (fault="underflow", operation="BigNumber.from", value=211.32064427312778, code=NUMERIC_FAULT, version=bignumber/5.0.14)
```

File: src/utils/MarketUtils.ts [hunk 2]
```diff
@@ -337,3 +363,457 @@ export const getDailyPriceHistory = async (group: string, synthMetadata: Record<
...
+ */
+export const getMiningRewards = async (
+  assetName: string,
+  asset: ISynth,
+  assetPrice: number,
+  cr: number,
+  tokenCount: number,
+) => {
+  // TODO Use params for setup instead of test setup
+  const rpcURL = 'https://fee7372b6e224441b747bf1fde15b2bd.eth.rpc.rivet.cloud';
```
Comment (ethedev) [inline src/utils/MarketUtils.ts]: to change using the same node url we use globally

File: src/utils/MarketUtils.ts [hunk 3]
```diff
@@ -337,3 +363,457 @@ export const getDailyPriceHistory = async (group: string, synthMetadata: Record<
...
+      }
+    } else if (assetName === "uSTONKS-APR21") {
+      if (current < week1Until) {
+        umaWeekRewards += 5000;
+        yamWeekRewards += 5000;
+      } else if (current < week2Until) {
+        umaWeekRewards += 10000;
+        yamWeekRewards += 10000;
+      }
+    }
```
Comment (ethedev) [inline src/utils/MarketUtils.ts]: this section is no longer needed

File: src/utils/MarketUtils.ts [hunk 4]
```diff
@@ -337,3 +363,457 @@ export const getDailyPriceHistory = async (group: string, synthMetadata: Record<
...
+    /* @ts-ignore */
+    const data = [
+      /* @ts-ignore */
+      assets["uGas"][1].emp.address,
+      /* @ts-ignore */
+      assets["uGas"][2].emp.address,
+      /* @ts-ignore */
+      assets["uGas"][3].emp.address,
+      /* @ts-ignore */
+      assets["uStonks"][0].emp.address,
```
Comment (ethedev) [inline src/utils/MarketUtils.ts]: we need to have here the new one added as well `assets["uStonks"][1].emp.address`

**Review by ethedev — CHANGES_REQUESTED** [review]
**Notes:**
1. on the landing page there are errors coming from coingecko eventhough there should not be any apr calculation initialized on there
2. initialization of the apr should happen on the explore page, or on the synth page if and whenever we display the apr value at, a check can be triggered if the data is fetched or yet to proceed accordingly

@byterose both of these points can be added/updated with the help of @sali0, after you finalize the apr calculation

File: src/utils/MarketUtils.ts (hunk 3 above)
Comment (ghost) [reply src/utils/MarketUtils.ts]: I will modify this after we discussed the uPUNKS rewards.

File: src/utils/MarketUtils.ts [hunk 5]
```diff
@@ -14,9 +14,35 @@ import {
...
+    sessionStorage.clear();
+  }
+} else {
+  sessionStorage.setItem("timestamp", Math.floor(Date.now() / 1000).toString());
+}
 
-sessionStorage.clear();
+const WETH = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+const YAM = '0x0AaCfbeC6a24756c20D41914F2caba817C0d8521';
+const UMA = '0x04fa0d235c4abf4bcf4787af4cf447de572ef828';
```
Comment (ind-igo) [inline src/utils/MarketUtils.ts]: remove

**Review by ind-igo — APPROVED** [review]

**Pull request merged**